lib/matplotlib/animation.py

```diff
@@ -296,8 +297,11 @@ def __init__(self, fps=5, codec=None, bitrate=None, extra_args=None,
 
         super().__init__(fps=fps, metadata=metadata, codec=codec,
                          bitrate=bitrate)
-
-        self.frame_format = 'rgba'
+        # The rcParam only applies to FileMovieWriters, but should be read here
+        # because setting to frame_format to 'rgba' would otherwise spuriously
+        # warn that that format is unsupported e.g. for FFMpegFileWriter.
+        self.frame_format = (mpl.rcParams['animation.frame_format']
+                             if isinstance(self, FileMovieWriter) else 'rgba')
```

Comment:
A base class that checks for a specific subclass? Ewww. There *has* to be a better way to accomplish this. Set `_frame_format` to `None` by default and do this in the getter on `None`?

Reply:
While that's doable, note that the fundamental problem here is that FileMovieWriter should not really inherit from MovieWriter: there's a bunch of methods where the FileMovieWriter override doesn't call the `super()` method (`setup`, `_frame_sink`, `grab_frame`); this PR just shows that likewise there's `__init__` functionality that isn't really fully inherited.  I guess the proper approach may be to move some of the common functionality to AbstractMovieWriter and make both MW and FMW directly inherit from AMW instead?...

Reply:
I think you can revert this with #19056?

Reply:
yup